runtime/bcutil/cfreader.c

```diff
@@ -2201,8 +2205,10 @@ checkClass(J9PortLibrary *portLib, J9CfrClassFile* classfile, U_8* segment, U_32
 		goto _errorFound;
 	}
 
-	/* check annotations and interfaces for invalid flags */
-	if ((value & CFR_ACC_INTERFACE) && (value & ~CFR_INTERFACE_CLASS_ACCESS_MASK))
+	/* check annotations and interfaces for invalid flags. If the ACC_INTERFACE flag is set,
+	 * the ACC_FINAL, ACC_SUPER, and ACC_ENUM flags set must not be set.
+	 */
+	if (J9_ARE_ALL_BITS_SET(value, CFR_ACC_INTERFACE) && J9_ARE_ANY_BITS_SET(value, CFR_INTERFACE_FORBIDDEN_ACCESS_MASK))
```

Comment:
ACC_MODULE is also invalid for Java 9

Reply:
Thanks. Looks like the Java 9 spec I've been using is a bit out of date.